### PR TITLE
Prepare FPWD for Controller Document specification.

### PR DESCRIPTION
**Pull request by msporny**
This PR readies the FPWD publication of the Controller Documents v1.0 specification (which is effectively a generalized and updated form of the DID Document format). It contains PRs #1, #2, and #3 authored by @selfissued. It does not contain PR #4, which removes all of the authors of the DID Core specification, upon which this document is based. It also contains a number of changes that were needed to fix up all of the broken links and references in the specification.

A static version of the document can be found here:

https://w3c.github.io/vc-controller-document/FPWD/2024-FPWD/

### CODE REVIEW

**Comment by selfissued** [comment]
@msporny, since you agree with the changes in #1 #2 and #3, let's merge them now.  Then it will be more obvious what this PR does, at which point I'll review it.  Right now, it's all jumbled together here.  Thanks.

**Comment by msporny** [comment]
> @msporny, since you agree with the changes in #1 #2 and #3, let's merge them now. Then it will be more obvious what this PR does, at which point I'll review it. Right now, it's all jumbled together here. Thanks.

WG process is to allow 7 days for people to vet each PR before it appears in the mainline specification. I can't merge any of those w/o violating WG process. 

In any case, it doesn't matter for the FPWD, since they're merged in the static FPWD copy (they are in there). Once we publish the FPWD, we can merge this PR, and all of it will go in at that point in time (which will be after the 7 day merge window).

**Comment by selfissued** [comment]
> WG process is to allow 7 days for people to vet each PR before it appears in the mainline specification. I can't merge any of those to main w/o violating WG process.

I understand, but arguably, these are all editorial, since they're applying changes that were already in the VC-JOSE-COSE or VC-DATA-INTEGRITY versions of the Controller document text.  Let's talk about this on the editors' call tomorrow.

> In any case, it doesn't matter for the FPWD, since they're merged in the static FPWD copy (they are in there). 

Yes, I believe you, but it still doesn't give us a clean diff that shows what changes are unique to this PR and which came from mine.  That's what I'm asking for, in order to efficiently review this PR.  Thanks.

**Comment by msporny** [comment]
> it still doesn't give us a clean diff that shows what changes are unique to this PR and which came from mine. That's what I'm asking for, in order to efficiently review this PR.

You can shift-click a range of commits in a PR to just view the changes that are on top of yours. To do this:

1. Go to "Files Changed" in a PR.
2. Click on the "Changes from all commits" dropdown toward the top-left of the diff view.
3. Hold down shift, click on the first commit you want to view, then click on the last commit in the range you want to view. 

Here's the URL that does that for this PR:

https://github.com/w3c/vc-controller-document/pull/11/files/9b77cfc1f21bb4fe92e08f67cbbb42e590c23a9b..40c2c6988838757672751db0bf66893d75db6a41

**Comment by iherman** [comment]
I wonder whether the publication of this FPWD should not be accompanied by PR's that _remove_ the corresponding sections from the JOSE-COSE and the DI documents. This would make it much clearer to the reviewers what is happenning here and would avoid discussions about the necessity of the new document.

**Comment by msporny** [comment]
> I wonder whether the publication of this FPWD should not be accompanied by PR's that _remove_ the corresponding sections from the JOSE-COSE and the DI documents. This would make it much clearer to the reviewers what is happening here and would avoid discussions about the necessity of the new document.

Strong -1 to that approach UNTIL vc-controller-document reaches Candidate Recommendation. Removing the controller document sections from the JOSE-COSE and DI specs is major surgery. We will need to continue to align everything in vc-controller-document, and we might hit a point at which there will be formal objections to move it to the Candidate Recommendation stage. If that happens, then we'll have to scramble to put the sections back into each document.

We are introducing a critical dependency at quite an unfortunate time (right at the end). We shouldn't tie all of these things together until we're sure they're all going to make it. At present, I give vc-controller-document a 33% chance of failure... while it's not a big chance, it is still a gamble at this point in time. The details will matter, and that's where we might see formal objections.

**Review by dlongley — APPROVED** [review]

**Review by decentralgabe — COMMENTED** [review]
I would rather wait for #1 #2 and #3 to be merged (and #4 to find consensus) before going forward with this

**Comment by brentzundel** [comment]
The starting point of this document needs to include all of the Controller Document features from both securing specs in order to comply with the resolution made by the group here:  https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-25-vcwg#resolution1

In order to accomplish that, this is the path I see before us:
1. merge PRs #1, #2, and #3 right away
2. merge this PR right after
3. present this document to the VCWG for approval to move to FPWD.
4. raise a PR updating the content from Data Integrity to match what is currently in that document and merge it

I believe that doing these things is necessary to meet the WG's consensus decision to create a Controller Document Spec. 

Once we have the Controller Document in that state, anyone who would like to see changes in the contents of the document should raise issues and our normal VCWG work mode will be in full effect, i.e., all issues will need to be addressed and PRs for resolving PRs will need our standard review period, etc.

**Comment by brentzundel** [comment]
@msporny I resolved conflicts in this PR in [1c1440f](https://github.com/w3c/vc-controller-document/pull/11/commits/1c1440f31c330bc0f810e0b8cb5c011a217f6caa)
Please let me know if I screwed things up and I will revert that commit.

File: index.html
Comment (selfissued) [inline index.html:1963]: `proofPurpose` isn't defined by this document, so I suggest removing this clause.

File: index.html
Comment (selfissued) [inline index.html:1973]: Per my earlier comment, `proofPurpose` isn't defined by this document.  This line seems vc-data-integrity specific.  Can we remove it for now and take up `proofPurpose` in its own issue?

File: index.html
Comment (selfissued) [inline index.html:2113]: Same comment as above.  Let's take up proofPurpose in a separate issue.

**Review by selfissued — COMMENTED** [review]
Let's take up `proofPurpose` in its own issue rather than adding it now.  Other than that, I'm good with these changes.

**Comment by brentzundel** [comment]
@selfissued per [my comment  here](https://github.com/w3c/vc-controller-document/pull/11#issuecomment-2096886899), please raise issues for your concerns so they can be properly tracked once we're past FPWD.

This PR should be merged as soon as @msporny confirms that it still contains his changes after my attempt to resolve conflicts.

**Review by selfissued — APPROVED** [review]
Approved, per @brentzundel's note at https://github.com/w3c/vc-controller-document/pull/11#issuecomment-2097032690.

I'll plan to file an issue in which we can discuss `proofPurpose` after this is merged.

**Comment by msporny** [comment]
Merging per Chair request to address WG consensus to get to FPWD in https://github.com/w3c/vc-controller-document/pull/11#issuecomment-2097032690.

**Pull request merged**
